packages/utils/src/ClipboardUtils.ts

```diff
@@ -38,3 +40,42 @@ export function copyToClipboardExecCommand(text: string): void {
 
   if (!result) throw new Error('Unable to execute copy command');
 }
+
+/**
+ * Reads text from the clipboard.
+ * @returns Promise that resolves to text from clipboard as string, or null if permissions are not supported/granted.
+ */
+export async function readFromClipboard(): Promise<string | null> {
```

Comment:
Instead of returning `string | null`, you should throw an error if the permissions are invalid. It's more descriptive and you could throw different kinds of errors. For this case, I'd create a `ClipboardPermissionsDeniedError` and `ClipboardUnavailableError` type (see `PasteError` for an example of creating a permissions type).
Then when calling `readFromClipboard`, wrap it in a `try/catch` (in IrisGridContextMenuHandler) and you can display an appropriate error message based on the type of error or check that it throws as expected (from the tests).